docs/diagrams/DeadlineSequenceDiagram.puml

```diff
@@ -0,0 +1,151 @@
+participant ":CommandResult" as CommandResult LOGIC_COLOR
+participant ":ParserUtil" as ParserUtil LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+participant "deadlines:DeadlineList" as DeadlineList MODEL_COLOR
+participant "personToAddDeadline:Person" as Person1 MODEL_COLOR
+participant "editedPerson:Person" as Person2 MODEL_COLOR
+end box
+
+[-> LogicManager : execute("deadline 1 /d return book 1/1/2023")
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand("deadline 1 /d return book 1/1/2023")
+activate AddressBookParser
+
+create DeadlineCommandParser
+AddressBookParser -> DeadlineCommandParser
+activate DeadlineCommandParser
+
+DeadlineCommandParser --> AddressBookParser
+deactivate DeadlineCommandParser
+
+AddressBookParser -> DeadlineCommandParser : parse("1 /d return book 1/1/2023")
+activate DeadlineCommandParser
+
+DeadlineCommandParser -> ParserUtil : parseIndex("1")
+activate ParserUtil
+
+ParserUtil --> DeadlineCommandParser : index
+deactivate ParserUtil
+
+DeadlineCommandParser -> ParserUtil : parseDeadlines([" return book 1/1/2023"])
+activate ParserUtil
+
+create DeadlineList
+ParserUtil -> DeadlineList : DeadlineList(deadlineList)
+activate DeadlineList
+
+DeadlineList --> ParserUtil : deadlines
+deactivate DeadlineList
+
+ParserUtil --> DeadlineCommandParser : deadlines
+deactivate ParserUtil
+
+create DeadlineCommand
+DeadlineCommandParser -> DeadlineCommand : DeadlineCommand(index, deadlines)
+activate DeadlineCommand
+
+DeadlineCommand --> DeadlineCommandParser : command
+deactivate DeadlineCommand
+
+DeadlineCommandParser --> AddressBookParser : command
+deactivate DeadlineCommandParser
+
+DeadlineCommandParser -[hidden]-> AddressBookParser
+destroy DeadlineCommandParser
+
+AddressBookParser --> LogicManager : command
+deactivate AddressBookParser
+
+LogicManager -> DeadlineCommand : execute(model)
+activate DeadlineCommand
+
+DeadlineCommand -> Model : getFilteredPersonList()
+activate Model
+
+Model --> DeadlineCommand : lastShownList
+deactivate Model
+
+DeadlineCommand -> Person1 : getName()
+activate Person1
+
+Person1 --> DeadlineCommand : name
+deactivate Person1
+
+DeadlineCommand -> Person1 : getPhone()
+activate Person1
+
+Person1 --> DeadlineCommand : phone
+deactivate Person1
+
+DeadlineCommand -> Person1 : getEmail()
+activate Person1
+
+Person1 --> DeadlineCommand : email
+deactivate Person1
+
+DeadlineCommand -> Person1 : getAddress()
+activate Person1
+
+Person1 --> DeadlineCommand : address
+deactivate Person1
+
+DeadlineCommand -> Person1 : getNotes()
+activate Person1
+
+Person1 --> DeadlineCommand : notes
+deactivate Person1
+
+DeadlineCommand -> Person1 : getTags()
+activate Person1
+
+Person1 --> DeadlineCommand : tags
+deactivate Person1
+
+DeadlineCommand -> Person1 : getFavouriteStatus()
+activate Person1
+
+Person1 --> DeadlineCommand : favourite
+deactivate Person1
```

Comment:
I think we can ignore some of the low level details :laughing:. I think the sequence diagram is too big as it is right now.

Reply:
Yeah, maybe I'll get rid of the huge chunk getting the personToAddDeadline's attributes.